OpenImis.ModulesV3/InsureeModule/Repositories/FamilyRepository.cs

```diff
@@ -356,5 +363,47 @@ public int GetOfficerIdByUserUUID(Guid userUUID)
                 throw e;
             }
         }
+
+        public int UpdateControlNumber(EnrollFamilyModel familyModel, NewFamilyResponse serverResponse)
+        {
+
+            foreach (var family in familyModel.Family)
+            {
+                foreach (var policy in family.Policies)
+                {
+                    var policyId = serverResponse.Family.Where(f => f.FamilyId == family.FamilyId).FirstOrDefault().Policies.Where(p => p.PolicyId == policy.PolicyId).Select(p => p.PolicyDBId).FirstOrDefault();
```

Comment:
This works for CHF when you have the CN in the payload. If we don't use ePayment in other contexts, we should first check if there is a CN in the Policy (I expect to be null for other implementations) and then do all SQL.